### PR TITLE
doc and licensing

### PR DESCRIPTION
**Pull request by NicolasDP**

### CODE REVIEW

File: cardano/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "cardano"
 version = "0.1.0"
-authors = ["vincent.hanquez@iohk.io", "nicolas.diprima@iohk.io"]
+authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>", "The Rust-Crypto Project Developers"]
 license = "MIT"
 description = "A Rust implementation of cardano types and wallet"
```
Comment (nahern) [inline cardano/Cargo.toml]: A Rust implementation of Cardano types and wallet

File: cardano/Cargo.toml [hunk 2]
```diff
@@ -2,8 +2,14 @@
 name = "cardano"
 version = "0.1.0"
 authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>", "The Rust-Crypto Project Developers"]
-license = "MIT"
-description = "A Rust implementation of cardano types and wallet"
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/input-output-hk/rust-cardano"
+homepage = "https://github.com/input-output-hk/rust-cardano/cardano#README.md"
+description = """
+The cardano library. It includes all the support for the cardano blockchains and
```
Comment (nahern) [inline cardano/Cargo.toml]: The Cardano library including full support for the Cardano blockchains and wallets.

File: cardano/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+[![Build Status](https://travis-ci.org/input-output-hk/rust-cardano.svg?branch=master)](https://travis-ci.org/input-output-hk/rust-cardano)
+![MIT or APACHE-2 licensed](https://img.shields.io/badge/licensed-MIT%20or%20APACHE--2-blue.svg)
+![Cardano Mainnet](https://img.shields.io/badge/Cardano%20Ada-mainnet-brightgreen.svg)
+![Cardano Staging](https://img.shields.io/badge/Cardano%20Ada-staging-brightgreen.svg)
+![Cardano Testnet](https://img.shields.io/badge/Cardano%20Ada-testnet-orange.svg)
+
+Crate with all the primitives for the [Cardano](https://www.cardano.org) blockchain:
+
+* blockchain validation;
+* HDWallet support (Icarus, Daedalus or customs);
```
Comment (nahern) [inline cardano/README.md]: HDWallet support (Icarus, Daedalus or custom wallet)

File: cardano/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+* transaction engine: input selection, fee estimation, signing and validation;
+
+This crates is aimed to support as many architectures as possible with as little dependencies as possible.
```
Comment (nahern) [inline cardano/README.md]: This crates package aims to support as many architectures as possible with as little dependencies as possible.

File: cardano/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+## supported compiler versions
+
+| Rust    | `test` |
+|---------|:------:|
+| stable  |   ✓    |
+| beta    |   ✓    |
+| nightly |   ✓    |
+
+We will always aim to support the current stable. However it is likely
+that older version of the rust compiler is supported too.
```
Comment (nahern) [inline cardano/README.md]: We will always aim to support the current stable version. However, it is likely that an older version of the rust compiler is also supported.

File: cardano/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+# License
+
+This project is licensed under either of
```
Comment (nahern) [inline cardano/README.md]: This project is licensed under either of the following licences:

File: cardano/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+ * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+   http://www.apache.org/licenses/LICENSE-2.0)
+ * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+   http://opensource.org/licenses/MIT)
+
+at your option.
```
Comment (nahern) [inline cardano/README.md]: 'at your option.' - would remove this text? seems incomplete.

File: cbor_event/README.md [hunk 8]
```diff
@@ -9,14 +12,56 @@ This crate has 0 dependency (and should not need any in the future). This
 is a design choice in order to guarantee as much as possible of the
 compatibility across multiple platform.
```
Comment (nahern) [inline cbor_event/README.md]: While some libraries provide an intermediate type representation, @@ -9,14 +12,56 @@ This crate has zero dependencies (and should not need any in the future). This is a design choice in order to guarantee as much compatibility as possible across multiple platforms.

File: cbor_event/README.md [hunk 9]
```diff
@@ -9,14 +12,56 @@ This crate has 0 dependency (and should not need any in the future). This
...
+## supported compiler versions
+
+| Rust    | `test` |
+|---------|:------:|
+| stable  |   ✓    |
+| beta    |   ✓    |
+| nightly |   ✓    |
+
+We will always aim to support the current stable. However it is likely
+that older version of the rust compiler is supported too.
```
Comment (nahern) [inline cbor_event/README.md]: We will always aim to support the current stable version. However, it is likely
that an older version of the rust compiler is also supported.

File: cbor_event/README.md [hunk 10]
```diff
@@ -9,14 +12,56 @@ This crate has 0 dependency (and should not need any in the future). This
...
+
+# License
+
+This project is licensed under either of
```
Comment (nahern) [inline cbor_event/README.md]: This project is licensed under either of the following licences:

File: cbor_event/README.md [hunk 11]
```diff
@@ -9,14 +12,56 @@ This crate has 0 dependency (and should not need any in the future). This
...
+
+ * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+   http://www.apache.org/licenses/LICENSE-2.0)
+ * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+   http://opensource.org/licenses/MIT)
+
+at your option.
```
Comment (nahern) [inline cbor_event/README.md]: would remove 'at your option' 

Possibly add 'Please choose which licence to use.'

File: protocol/Cargo.toml [hunk 12]
```diff
@@ -2,8 +2,14 @@
...
 authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>", "The Rust-Crypto Project Developers"]
-license = "MIT"
-description = "protocol"
+license = "MIT OR Apache-2.0"
+readme = "README.md"
+repository = "https://github.com/input-output-hk/rust-cardano"
+homepage = "https://github.com/input-output-hk/rust-cardano/protocol#README.md"
+description = """
+Cardano network protocol. Provides implementation to fetch blocks from a remote and
+to send transactions to a remote.
```
Comment (nahern) [inline protocol/Cargo.toml]: Cardano network protocol. Provides an implementation to fetch blocks from a remote **location/node** and
to send transactions to a remote **location/node**.

File: protocol/README.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
+# Cardano network Protocol
+
+[![Build Status](https://travis-ci.org/input-output-hk/rust-cardano.svg?branch=master)](https://travis-ci.org/input-output-hk/rust-cardano)
+![MIT or APACHE-2 licensed](https://img.shields.io/badge/licensed-MIT%20or%20APACHE--2-blue.svg)
+![Cardano Mainnet](https://img.shields.io/badge/Cardano%20Ada-mainnet-brightgreen.svg)
+![Cardano Staging](https://img.shields.io/badge/Cardano%20Ada-staging-brightgreen.svg)
+![Cardano Testnet](https://img.shields.io/badge/Cardano%20Ada-testnet-orange.svg)
+
+Support for the [Cardano](https://www.cardano.org) network protocol:
```
Comment (nahern) [inline protocol/README.md]: Support for the [Cardano](https://www.cardano.org) network protocol to perform the following functions:

File: protocol/README.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+* fetch blocks;
+* send transactions.
```
Comment (nahern) [inline protocol/README.md]: remove ';' and '.' at end of lines?

File: protocol/README.md [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+## supported compiler versions
+
+| Rust    | `test` |
+|---------|:------:|
+| stable  |   ✓    |
+| beta    |   ✓    |
+| nightly |   ✓    |
+
+We will always aim to support the current stable. However it is likely
+that older version of the rust compiler is supported too.
```
Comment (nahern) [inline protocol/README.md]: We will always aim to support the current stable version. However, it is likely that an older version of the Rust compiler is also supported.

File: protocol/README.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+# License
+
+This project is licensed under either of
```
Comment (nahern) [inline protocol/README.md]: This project is licensed under either of the following licences:

File: protocol/README.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+ * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+   http://www.apache.org/licenses/LICENSE-2.0)
+ * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+   http://opensource.org/licenses/MIT)
+
+at your option.
```
Comment (nahern) [inline protocol/README.md]: remove 

replace with 'Please choose licence you want to use.'

File: cardano/Cargo.toml [hunk 18]
```diff
@@ -2,8 +2,14 @@
...
+wallets. It also provides helpers to maintain HDWallets and includes a transaction
+build engine.
```
Comment (nahern) [inline cardano/Cargo.toml]: The Cardano library including full support for the Cardano blockchains and wallets. It also provides helpers to maintain HDWallets and includes a transaction build engine.

File: cardano/Cargo.toml [hunk 19]
```diff
@@ -2,8 +2,14 @@
 name = "cardano"
 version = "0.1.0"
 authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>", "The Rust-Crypto Project Developers"]
-license = "MIT"
-description = "A Rust implementation of cardano types and wallet"
```
Comment (nahern) [inline cardano/Cargo.toml]: A Rust implementation of Cardano types and wallets.

File: cbor_event/Cargo.toml [hunk 20]
```diff
@@ -2,12 +2,11 @@
...
-license = "MIT"
+license = "MIT OR Apache-2.0"
 readme = "README.md"
 categories = ["binary", "cbor", "serialisation", "deserialisation", "cardano", "wasm"]
 repository = "https://github.com/input-output-hk/rust-cardano"
-homepage = "https://github.com/input-output-hk/rust-cardano"
-# documentation = "https://docs.rs/wasm-bindgen"
+homepage = "https://github.com/input-output-hk/rust-cardano/cbor_event#README.md"
 description = """
 CBOR Event library, no dependencies, all platform and wasm compatible.
```
[the remark this answers is not in the document]
Comment (NicolasDP) [reply cbor_event/Cargo.toml]: what do you mean by:

> **Readme file for the** CBOR Event library. There are no dependencies, all platforms are supported and it is wasm compatible.

File: cryptoxide/README.md [hunk 21]
```diff
@@ -0,0 +1,79 @@
+# Cryptoxide
+
+[![Build Status](https://travis-ci.org/input-output-hk/rust-cardano.svg?branch=master)](https://travis-ci.org/input-output-hk/rust-cardano)
+![MIT or APACHE-2 licensed](https://img.shields.io/badge/licensed-MIT%20or%20APACHE--2-blue.svg)
+
+A pure-rust implementation of various modern cryptographic algorithms, which no dependencies
+and no foreign code (specially C or assembly), to work with WASM and embedded devices.
```
Comment (nahern) [inline cryptoxide/README.md]: A pure Rust implementation of various modern cryptographic algorithms, which has no dependencies
and no foreign code (specially C or assembly code). This is compatible with WASM and embedded devices.

File: cryptoxide/README.md [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+
+This crates is aiming to support as many architectures as possible with as
+little dependencies as possible.
```
Comment (nahern) [inline cryptoxide/README.md]: This crates package aims to support as many architectures as possible with as
little dependencies as possible.

File: cryptoxide/README.md [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+
+Disclaimer: As everything cryptographically related, no warranties.
```
Comment (nahern) [inline cryptoxide/README.md]: Disclaimer: There are no warranties in use as everything is cryptographically-related.

File: cryptoxide/README.md [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+
+## Fork information
+
+This is a fork of [Rust-Crypto by DaGenix](https://github.com/DaGenix/rust-crypto),
+which we owe a debt of gratitude for starting some good quality pure rust implementations
+of various cryptographic algorithms.
```
Comment (nahern) [inline cryptoxide/README.md]: ## Fork information
This is a fork of [Rust-Crypto by DaGenix](https://github.com/DaGenix/rust-crypto), to
which we owe a debt of gratitude for starting some good quality pure Rust implementations
of various cryptographic algorithms.

File: cryptoxide/README.md [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+Notable Differences with the original sources:
```
Comment (nahern) [inline cryptoxide/README.md]: Notable differences with the original sources:

File: cryptoxide/README.md [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+## supported compiler versions
+
+| Rust    | `test` |
+|---------|:------:|
+| stable  |   ✓    |
+| beta    |   ✓    |
+| nightly |   ✓    |
+
+We will always aim to support the current stable. However it is likely
+that older version of the rust compiler is supported too.
```
Comment (nahern) [inline cryptoxide/README.md]: We will always aim to support the current stable version. However, it is likely that an older version of the Rust compiler is also supported.

File: cryptoxide/README.md [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+
+# License
+
+This project is licensed under either of
```
Comment (nahern) [inline cryptoxide/README.md]: This project is licensed under either of the following licences:

File: cryptoxide/README.md [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+
+ * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+   http://www.apache.org/licenses/LICENSE-2.0)
+ * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+   http://opensource.org/licenses/MIT)
+
+at your option.
```
Comment (nahern) [inline cryptoxide/README.md]: remove 'at your option'

replace with 'Please choose the licence you want to use.'

File: storage/Cargo.toml [hunk 29]
```diff
@@ -2,6 +2,12 @@
 name = "storage"
 version = "0.1.0"
 authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>"]
+keywords = [ "Cardano", "Storage", "Immutable", "FileSystem" ]
+description = """
+The cardano storage library. It provides 2 different levels of functionality, from lowlevel
+fast immutable or append only file without any assumption about the data, to a higher level
+intertwined with the cardano blockchain
```
Comment (nahern) [inline storage/Cargo.toml]: The Cardano storage library. It provides two different levels of functionality; from low level
fast immutable or append only file without any assumption about the data, to a higher level that is 
intertwined with the Cardano blockchain itself.

File: exe-common/Cargo.toml [hunk 30]
```diff
@@ -2,6 +2,14 @@
 name = "exe-common"
 version = "0.1.0"
 authors = ["Vincent Hanquez <vincent.hanquez@iohk.io>", "Nicolas Di Prima <nicolas.diprima@iohk.io>", "The Rust-Crypto Project Developers"]
+license = "MIT OR Apache-2.0"
+readme = "README.md"
+repository = "https://github.com/input-output-hk/rust-cardano"
+homepage = "https://github.com/input-output-hk/rust-cardano/exe-common#README.md"
+description = """
+Helpers and toolings for the cardano protocol, keeping blockchain synced and storing
+the blockchain locally.
```
Comment (nahern) [inline exe-common/Cargo.toml]: Helpers and tools for the Cardano protocol. This includes keeping the blockchain synced and storing the blockchain locally.

File: exe-common/README.md [hunk 31]
```diff
@@ -0,0 +1,56 @@
+# Cardano helpers and tooling
+
+[![Build Status](https://travis-ci.org/input-output-hk/rust-cardano.svg?branch=master)](https://travis-ci.org/input-output-hk/rust-cardano)
+![MIT or APACHE-2 licensed](https://img.shields.io/badge/licensed-MIT%20or%20APACHE--2-blue.svg)
+![Cardano Mainnet](https://img.shields.io/badge/Cardano%20Ada-mainnet-brightgreen.svg)
+![Cardano Staging](https://img.shields.io/badge/Cardano%20Ada-staging-brightgreen.svg)
+![Cardano Testnet](https://img.shields.io/badge/Cardano%20Ada-testnet-orange.svg)
+
+Extra toolings and helpers for the [Cardano](https://www.cardano.org) network protocol:
```
Comment (nahern) [inline exe-common/README.md]: Extra tools and helpers for the [Cardano](https://www.cardano.org) network protocol:

File: exe-common/README.md [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+## supported compiler versions
+
+| Rust    | `test` |
+|---------|:------:|
+| stable  |   ✓    |
+| beta    |   ✓    |
+| nightly |   ✓    |
+
+We will always aim to support the current stable. However it is likely
+that older version of the rust compiler is supported too.
```
Comment (nahern) [inline exe-common/README.md]: We will always aim to support the current stable version. However, it is likely that an older version of the Rust compiler is also supported.

File: exe-common/README.md [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+
+# License
+
+This project is licensed under either of
```
Comment (nahern) [inline exe-common/README.md]: This project is licensed under either of the following licences:

File: exe-common/README.md [hunk 34]
```diff
@@ -0,0 +1,56 @@
...
+
+ * Apache License, Version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+   http://www.apache.org/licenses/LICENSE-2.0)
+ * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+   http://opensource.org/licenses/MIT)
+
+at your option.
```
Comment (nahern) [inline exe-common/README.md]: remove 'at your option'

replace with 'Please choose the licence you want to use.'

**Pull request merged**